sys/build.sh

```diff
@@ -1,11 +1,54 @@
 #!/bin/sh
 
+GetPlatform() {
+	# Get OS and platform to decide if we need to limit memory usage
+	# during the build
+	PLATFORM=$(uname -a)
+	case "$PLATFORM" in
+	"Linux raspberrypi"*) MAX_MEM_PER_JOB=300000;;
+	"Linux"*) MAX_MEM_PER_JOB=150000;;
+	*) MAX_MEM_PER_JOB=200000 # If platform is not Linux (fallback value)
+	esac
+}
+
+BuildJobsThrottler(){
+	echo "Building on Linux : computing number of allowed parallel jobs."
+	echo "Maximum allowed RAM memory per job is $MAX_MEM_PER_JOB kB."
+
+	# Get number of CPUs on this target
+	# getconf does not exit on Darwin. Use sysctl on Darwin machines.
+	CPU_N=$(getconf _NPROCESSORS_ONLN 2>/dev/null || sysctl -n hw.ncpu) 
+	printf "Number of CPUs is %s and "  "$CPU_N"
+
+	# Get remaining RAM that could be used for this build
+	FREE_RAM=$(grep MemAvailable /proc/meminfo | sed 's/[^0-9]//g')
```

Comment:
This doesn't work on Mac os btw

Reply:
Hi! 
I don't have access to a Mac, would you be able to help?
I tested this on rasbian, Debian and Fedora.